lib/transport/selenium-webdriver/method-mappings.js

```diff
@@ -124,76 +124,133 @@ module.exports = class MethodMappings {
+          };
+        },
 
-        getAllWindowHandles: '/window_handles',
+        async getAllWindowHandles() {
+          const value = await this.driver.getAllWindowHandles();
 
-        getWindowPosition(windowHandle) {
-          return `/window/${windowHandle}/position`;
+          return {
+            value
+          };
         },
 
-        maximizeWindow(windowHandle) {
-          return TransportActions.post(`/window/${windowHandle}/maximize`);
+        async getWindowPosition() {
+          const { x, y } = await this.driver.manage().window().getRect();
+
+          return {
+            value: {
+              x,
+              y
+            }
+          };
         },
 
-        minimizeWindow() {
-          return TransportActions.post('/window/minimize');
+        async maximizeWindow(windowHandle) {
+          await this.driver.manage().window().maximize();
+
+          return {
+            value: null
+          };
         },
 
-        fullscreenWindow() {
-          return TransportActions.post('/window/fullscreen');
+        async minimizeWindow() {
+          await this.driver.manage().window().minimize();
+
+          return {
+            value: null
+          };
         },
 
-        openNewWindow() {
-          return TransportActions.post('/window/new');
+        async fullscreenWindow() {
+          await this.driver.manage().window().fullscreen();
+
+          return {
+            value: null
+          };
         },
 
-        setWindowPosition(windowHandle, offsetX, offsetY) {
-          return TransportActions.post({
-            path: `/window/${windowHandle}/position`,
-            data: {
-              x: offsetX,
-              y: offsetY
-            }
-          });
+        async openNewWindow() {
+          // return TransportActions.post('/window/new');
+          await this.driver.switchTo().newWindow('window');
+
+          return {
+            value: null
+          };
         },
 
-        getWindowSize(windowHandle) {
-          return `/window/${windowHandle}/size`;
+        //Kept windowHandle as params for backward comptability.
+        async setWindowPosition(windowHandle, offsetX, offsetY) {
+          await this.driver.manage().window().setRect({
+            x: offsetX,
+            y: offsetY
+          });
+
+          return {
+            value: null
+          };
         },
 
-        setWindowSize(windowHandle, width, height) {
-          return TransportActions.post({
-            path: `/window/${windowHandle}/size`,
-            data: {
-              width: width,
-              height: height
+        async getWindowSize() {
+          const { width, height } = await this.driver.manage().window().getRect();
+
+          return {
+            value: {
+              width,
+              height
             }
+          };
+        },
+
+        //Kept windowHandle as params for backward comptability.
+        async setWindowSize(windowHandle, width, height) {
```

Comment:
no need for backward compatibility here, since this is an internal protocol action.

Reply:
Actually Internal protocol action won't work if I remove `windowHandle` from params and if we change change protocol action that would lead to breaking changes in previous versions, so was confused.

Reply:
The windowHandle needs to be removed for w3c-compliant drivers because it doesn't exist anymore and so we need to update the nightwatch `windowSize` command to make it optional.